code/obj/submachine/slots.dm

```diff
@@ -93,10 +107,24 @@
 		if(src.scan)
 			boutput(user, "<span class='alert'>There is a card already in the slot machine.</span>")
 		else
+			var/obj/item/card/id/idcard = I
 			boutput(user, "<span class='notice'>You insert your ID card.</span>")
 			usr.drop_item()
 			I.set_loc(src)
+			if(!idcard.registered)
+				boutput(usr, "<span class='alert'>No account data found!</span>")
+				usr.put_in_hand_or_eject(I)
+				ui_interact(user)
+				return
+			var/enterpin = input(user, "Please enter your PIN number.", "Enter PIN", 0) as null|num
+			if (enterpin != idcard.pin)
+				boutput(user, "<span class='alert'>Pin number incorrect.</span>")
+				usr.put_in_hand_or_eject(I)
+				ui_interact(user)
+				return
+			boutput(user, "<span class='notice'>Card authorized.</span>")
```

Comment:
It'd be nice to see some code-reuse in `attackby()` and `ui_act()`

Reply:
Thanks, I've adjusted it so now ui_act calls the attackby function when inserting a card.